### ISSUE TITLE
Failed to revert a block during new install of V1.3.6

### ISSUE THREAD

**Issue by PulsarAV1** [bitcoinclassic/bitcoinclassic#302]
I'm not a developer, however I'm willing to assist in troubleshooting this issue.

Running Win7 64-bit fully patched.

Debug log:

22:32:09 [Net] parameter interaction: -whitelistforcerelay=1 -> setting -whitelistrelay=1 
    .366 [UAHF] UAHF state: 1 start time: 1501590000 
22:32:16 [Wallet] Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010) 
    .122 [Bitcoin] Startup time: "2017-09-02 22:32:16" 
    .122 [Bitcoin] Using data directory "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash" 
    .123 [Bitcoin] Using config file "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash\bitcoin.conf" 
    .129 [Wallet] Using wallet "wallet.dat" 
    .129 [Bitcoin] init message: "Verifying wallet..." 
    .150 [Bitcoin] init message: "Loading block index..." 
    .150 [DB] Opening LevelDB in "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash\blocks\index" 
    .237 [DB] Opening LevelDB in "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash\chainstate" 
22:32:29 [UAHF] The UAHF fork-block is not in the main chain 
    .934 [UAHF] Set UAHF: 2 
    .934 [UAHF] Detected your chain was not UAHF, will need to rollback to the fork-block! 
    .935 [UAHF]  Fork-block is at: 478559 
    .935 [Bitcoin] init message: "Rolling back chain to UAHF-branch-point (4648 blocks)..." 
    .935 [UAHF]  reverting 0000000000000000008569461653d9fbab260f401c5eb54514f39344b40a193c 483207 
22:35:37 [UAHF] Failed to revert a block, likely due to a database error. This is fatal. Shutting down now 
22:35:43 [Bitcoin] Shutdown requested. Exiting. 
    . 12 [HTTP] Interrupting HTTP server 
    . 14 [RPC] Interrupting HTTP RPC server 
    . 15 [RPC] Stopping HTTP RPC server 
    . 16 [HTTP] Stopping HTTP server 
    . 16 [HTTP] Stopped HTTP server

**Comment by zander** [comment]
If your node does pruning, then you can not use the rollback feature.


On Sunday, 3 September 2017 00:48:40 CEST PulsarAV1 wrote:
> I'm not a developer, however I'm willing to assist in troubleshooting this
> issue.
> 
> Running Win7 64-bit fully patched.
> 
> Debug log:
> 
> 22:32:09 [Net] parameter interaction: -whitelistforcerelay=1 -> setting
> -whitelistrelay=1 .366 [UAHF] UAHF state: 1 start time: 1501590000
> 22:32:16 [Wallet] Using BerkeleyDB version Berkeley DB 4.8.30: (April  9,
> 2010) .122 [Bitcoin] Startup time: "2017-09-02 22:32:16"
>     .122 [Bitcoin] Using data directory
> "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash" .123 [Bitcoin] Using
> config file "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash\bitcoin.conf"
> .129 [Wallet] Using wallet "wallet.dat"
>     .129 [Bitcoin] init message: "Verifying wallet..."
>     .150 [Bitcoin] init message: "Loading block index..."
>     .150 [DB] Opening LevelDB in
> "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash\blocks\index" .237 [DB]
> Opening LevelDB in
> "C:\Users\XXXXXXXX\AppData\Roaming\BitcoinCash\chainstate" 22:32:29
> [UAHF] The UAHF fork-block is not in the main chain
>     .934 [UAHF] Set UAHF: 2
>     .934 [UAHF] Detected your chain was not UAHF, will need to rollback to
> the fork-block! .935 [UAHF]  Fork-block is at: 478559
>     .935 [Bitcoin] init message: "Rolling back chain to UAHF-branch-point
> (4648 blocks)..." .935 [UAHF]  reverting
> 0000000000000000008569461653d9fbab260f401c5eb54514f39344b40a193c 483207
> 22:35:37 [UAHF] Failed to revert a block, likely due to a database error.
> This is fatal. Shutting down now 22:35:43 [Bitcoin] Shutdown requested.
> Exiting.
>     . 12 [HTTP] Interrupting HTTP server
>     . 14 [RPC] Interrupting HTTP RPC server
>     . 15 [RPC] Stopping HTTP RPC server
>     . 16 [HTTP] Stopping HTTP server
>     . 16 [HTTP] Stopped HTTP server


-- 
Tom Zander
Blog: https://zander.github.io
Vlog: https://vimeo.com/channels/tomscryptochannel

**Comment by ChrisPapaioannou** [comment]
I'm running what I believe to be a fully synchronised Bitcoin Core node, which hasn't been configured to prune (130 GB worth of blocks resident on disk and nothing in the bitcoin.conf indicating pruning). 

Today I shutdown the Core node, installed Classic 1.3.6 UAHF version and fired it up. It displayed a splash indicating it had created a separate Cash directory and then encountered a fatal error shortly after. My debug log is included below, which is extremely similar to PulsarAV1's. This is running on a Windows 10 x64 box. Happy to assist with any testing.

18:34:55 [Net] parameter interaction: -whitelistforcerelay=1 -> setting -whitelistrelay=1 
    .247 [UAHF] UAHF state: 1 start time: 1501590000 
18:34:56 [Wallet] Using BerkeleyDB version Berkeley DB 4.8.30: (April  9, 2010) 
    .578 [Bitcoin] Startup time: "2017-09-03 18:34:56" 
    .578 [Bitcoin] Using data directory "C:\Users\Chris\AppData\Roaming\BitcoinCash" 
    .578 [Bitcoin] Using config file "C:\Users\Chris\AppData\Roaming\BitcoinCash\bitcoin.conf" 
    .579 [Wallet] Using wallet "wallet.dat" 
    .579 [Bitcoin] init message: "Verifying wallet..." 
    .592 [Bitcoin] init message: "Loading block index..." 
    .592 [DB] Opening LevelDB in "C:\Users\Chris\AppData\Roaming\BitcoinCash\blocks\index" 
    .773 [DB] Opening LevelDB in "C:\Users\Chris\AppData\Roaming\BitcoinCash\chainstate" 
18:35:02 [UAHF] The UAHF fork-block is not in the main chain 
    .795 [UAHF] Set UAHF: 2 
    .795 [UAHF] Detected your chain was not UAHF, will need to rollback to the fork-block! 
    .795 [UAHF]  Fork-block is at: 478559 
    .796 [Bitcoin] init message: "Rolling back chain to UAHF-branch-point (4774 blocks)..." 
    .796 [UAHF]  reverting 000000000000000000a957f65a8897c5cca0981f745bbc3855e5cb4117b31f5c 483333 
18:35:06 [UAHF] Failed to revert a block, likely due to a database error. This is fatal. Shutting down now 
    .904 [Bitcoin] Shutdown requested. Exiting. 
    .916 [HTTP] Interrupting HTTP server 
    .916 [RPC] Interrupting HTTP RPC server 
    .916 [RPC] Stopping HTTP RPC server 
    .917 [HTTP] Stopping HTTP server 
    .917 [HTTP] Stopped HTTP server

**Comment by zander** [comment]
Which version of Core were you running?

The unreleased version of Core seems to have changed its database format so nothing but Core 0.15 can read it.

**Comment by ChrisPapaioannou** [comment]
Bitcoin Core v0.14.2 (64-bit). Since my last comment I launched the client with the -reindex flag and left it to complete. Upon completion I closed Core, opened Classic and got the same fatal error and debug.log output.

**Comment by PulsarAV1** [comment]
Sorry for the late reply. I've been out of range for communications.

I too have Core version 0.14.2.

My (painful) solution was to rename my Core directory and re-sync the
entire chain with Classic in a separate directory. I'm still a few days
away from completion.

PulsarAV


On Sep 4, 2017 3:22 PM, "Chris Papaioannou" <notifications@github.com>
wrote:

Bitcoin Core v0.14.2 (64-bit). Since my last comment I launched the client
with the -reindex flag and left it to complete. Upon completion I closed
Core, opened Classic and got the same fatal error and debug.log output.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/bitcoinclassic/bitcoinclassic/issues/302#issuecomment-327017850>,
or mute the thread
